### PR TITLE
compiler: allow to use `return` with no arguments

### LINKED ISSUE #929: Compiler throws panic on a function w/o return value
neo-go master-2.x branch
neo-go version 0.74.1-pre-148-ge097e86b

I got a `loop` function with `for` loop over slice, stored in the contract storage. 

```go
package foo

import "github.com/nspcc-dev/neo-go/pkg/interop/storage"

func loop(ctx storage.Context, key string, value int) {
	list := storage.Get(ctx, key).([]int)
	for i := 0; i < len(list); i++ {
		if list[i] == value {
			return
		}
	}
	list = append(list, value)
	storage.Put(ctx, key, list)
}

func Main() int {
	ctx := storage.GetContext()
	loop(ctx, "foo", 1)
	return 1
}
```

When I run this code in a tests, I expect to get a storage error, something like this one.
```
error encountered at instruction 71 (SYSCALL): failed to invoke syscall: could not find [102 111 111]
```

But instead compiler throws panic.

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
```

If I add return value in a `loop` function then it compiles correctly:

```go
func loop(ctx storage.Context, key string, value int) int {
	list := storage.Get(ctx, key).([]int)
	for i := 0; i < len(list); i++ {
		if list[i] == value {
			return 0
		}
	}
	list = append(list, value)
	storage.Put(ctx, key, list)
	return 0
}
```

Are there any special rules for functions without return values?

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #929 .

### CODE REVIEW

**Comment by fyrchik** [comment]
Don't merge it yet, there is more to do.

File: pkg/compiler/function_call_test.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ func TestFunctionWithVoidReturn(t *testing.T) {
 			return x + y
 		}
 
-		func getSomeInteger() { }
+		func getSomeInteger() { return }
```
Comment (roman-khimov) [inline pkg/compiler/function_call_test.go:125]: I think we need two tests anyway, one with explicit `return` and one without it.

File: pkg/compiler/codegen.go [hunk 2]
```diff
@@ -261,8 +261,10 @@ func (c *codegen) convertFuncDecl(file ast.Node, decl *ast.FuncDecl) {
 
 	ast.Walk(c, decl.Body)
 
-	// If this function returns the void (no return stmt) we will cleanup its junk on the stack.
-	if !hasReturnStmt(decl) {
+	// If we have reached the end of the function without encountering `return` statement,
+	// we should clean alt.stack manually.
+	// This can be the case with void and named-return functions. For now we support only former case.
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:266]: Do we have an issue for functions with return variables?

File: pkg/compiler/function_call_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/compiler/function_call_test.go]: Fixed.

File: pkg/compiler/codegen.go (hunk 2 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:266]: Not anymore :)

Comment (roman-khimov) [reply pkg/compiler/codegen.go:266]: Well, then this comment is a bit outdated now.

**Pull request merged**